aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,172 @@
+
+![Interpreted mode](index/_static/interpretedmode.png)
+
+Interpreted mode is similar to how the core internals of the desktop Common Language Runtime (CLR) work. The desktop CLR is distributed precompiled into native code, which then loads and executes .NET assembly files. One key difference is that the desktop CLR uses just-in-time (JIT) compilation extensively to make execution faster. Mono on WebAssembly is closer to a pure interpretation model.
+
+### Ahead-of-time (AOT) compilation mode
+
+In ahead-of-time (AOT) compilation mode, the app's .NET assemblies are transformed into WebAssembly binaries at build time. At runtime, there's no interpretation. The code executes immediately. It's still necessary to load part of the Mono runtime. Features for low-level .NET services are loaded, such as garbage collection. Features for parsing .NET assemblies aren't required and aren't loaded.
+
+![AOT mode](index/_static/aotmode.png)
+
+This is similar to how the [Ngen.exe tool](/dotnet/framework/tools/ngen-exe-native-image-generator) has historically allowed AOT compilation of .NET binaries into native machine code. More recently, CoreRT provides a complete native AOT .NET runtime for executing .NET binaries.
+
+## Interpreted versus AOT
+
+Interpreted mode provides a faster development cycle than AOT. When code is changed under the interpreted model, the app can be rebuilt and reloaded in the browser within a few seconds. An AOT rebuild might take several minutes to compile and reload. Because interpreted mode offers faster recompilation, it may be preferred for development. It remains an open question how to best apply interpreted and AOT modes of compilation in Blazor.
+
+## Browsers that don't support WebAssembly
+
+When the .NET runtime is compiled into WebAssembly, it's also compiled into [asm.js](https://en.wikipedia.org/wiki/Asm.js). *asm.js* is a subset of JavaScript and can be executed by JavaScript runtimes in browsers going back several years. When Blazor loads in the browser, it checks for WebAssembly support. If WebAssembly isn't supported, the *asm.js* runtime is loaded. *asm.js* isn't always used becuase it's larger and slower than the WebAssembly runtime.
+
+## Blazor, a SPA framework
+
+Blazor has a growing standard feature set to solve common app requirements, such as UI composition, state management, and routing. Features are designed around the strengths of .NET and the C# language with careful consideration given to tooling support.
+
+Blazor is inspired by the most popular SPA frameworks, such as [React](https://reactjs.org/), [Vue](https://vuejs.org/), and [Angular](https://angular.io/), plus some Microsoft technologies, such as [Razor Pages](xref:mvc/razor-pages/index). The goal is to provide SPA features that web developers have found most successful in existing frameworks in a way that capitalizes on .NET's strengths.
```

Comment:
While true, I'd prefer not to call out this part of Blazor's heritage here. It's not useful if you don't already know these frameworks, and we don't know what the future of these JavaScript frameworks will be.